### DISCUSSION TITLE
MDI (Material Design Icons)

### DISCUSSION THREAD

**Discussion by narrcom** [ExperienceLovelace/ha-floorplan#43] [Q&A] [answered]
Is there a way to change a picture element to a MDI icon with color without saving it locally..?

**Comment by exetico** [comment, marked as answer]
No, we have no supports of changing a element with an icon. ha-floorplan are "powered" by SVG-files.

With that said, if Home Assistnat exposes the icons so they're able to add with the :before attribute in a `content`-option, that should be possible. But this has nothing to do with ha-floorplan :-)  But... The SVG-icons are not exposed as selectors or fonts, in Home Assistant.

**Reply by narrcom** [reply]
no worries, excellent work btw... the name is a little misleading... it can display anything you want, can't wait for the documentation to be completed...

I discovered the following if it helps anyone...

**to get attributes for current entity...**
- entity.attributes.attribute_name
   eg. `entity.attributes.brightness`
- entity.attributes.attribute_array[attribute_index].attribute_name
   eg. `entity.attributes.forecast[0].temperature`

**To get state or attributes for other entity...**
- replace entity. with entities["domain.entity"].
   eg. `entities["light.bedroom"].attributes.brightness`
      or `entities["light.bedroom"].state`

**Reply by pkozul** [reply]
@narrcom 

Yep, that's correct. We will add this to the doco soon, as it's from the original version of floorplan.

> You have full access to the entity's state object, which allows you to access other properties such as `last_changed`, `attributes.friendly_name`, etc. The full set of objects available to your template literals is shown below:
> 
> - `entity` - the state object for the current entity
> - `entities` - the state objects for all entities
> - `hass` - the hass object
> - `config` - the floorplan configuration

**Reply by narrcom** [reply]
Excellent, do you have a list of internal services like floorplan.text_set or floorplan.image_set..?

**Reply by narrcom** [reply]
Also is there a way to align text, or numbers to the right..?

**Reply by pkozul** [reply]
> Excellent, do you have a list of internal services like floorplan.text_set or floorplan.image_set..?

In addition to being able to call any of the HA services, there are currently 4 floorplan-based services.:

- `floorplan.class_set`
- `floorplan.style_set`
- `floorplan.text_set`
- `floorplan.image_set`

**Reply by exetico** [reply]
> Also is there a way to align text, or numbers to the right..?

That's just CSS mate 😊👌 (text-align:right, or float, it depends on the SVG structure, for sure)

But it depends on the situation, for sure.

If you have a specific need, consider to create a separate question for it, with a detailed request including an example or so 😊🎉

**Reply by pkozul** [reply]
> Also is there a way to align text, or numbers to the right..?

I haven't tried aligning text within SVG files, but maybe this might help?

https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/text-anchor

**Reply by narrcom** [reply]
how can i apply the style when i already have a state action setting the text as follows.

```
# Today's Maximum Temperature
- entity: weather.home
  element: max_temp
  state_action:
    action: call-service
    service: floorplan.text_set
    service_data: '${entity.attributes.forecast[0].temperature}'
  tap_action: more-info
```

**Reply by OzGav** [reply]
You can just indent with a hyphen like this:
```
            - entities:
                - sensor.wupws_uv
              name: Colour Coded UV Sensors
              state_action:
                - service: floorplan.text_set
                  service_data: '${entity.state != "unavailable" ? Math.ceil(entity.state) : "?"}'
                - service: floorplan.class_set  
                  service_data: |
                    >
                      var temp = parseInt(entity.state); 
                      if (temp < 3) return "uv-low"; 
                      else if (temp < 8) return "uv-medium"; 
                      else return "uv-high";

```
